pkg/api/api.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2016 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"regexp"
+
+	"github.com/go-kit/kit/log"
+	"k8s.io/client-go/1.5/kubernetes"
+
+	"github.com/coreos/prometheus-operator/pkg/k8sutil"
+	"github.com/coreos/prometheus-operator/pkg/prometheus"
+)
+
+type API struct {
+	kclient *kubernetes.Clientset
+	pclient *prometheus.MonitoringClient
+	logger  log.Logger
+}
+
+func New(conf prometheus.Config, l log.Logger) (*API, error) {
+	cfg, err := k8sutil.NewClusterConfig(conf.Host, conf.TLSInsecure, &conf.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	kclient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	pclient, err := prometheus.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &API{
+		kclient: kclient,
+		pclient: pclient,
+		logger:  l,
+	}, nil
+}
+
+var (
+	prometheusRoute = regexp.MustCompile("/apis/monitoring.coreos.com/v1alpha1/namespaces/(.*)/prometheuses/(.*)/status")
+)
+
+func (api *API) Register(mux *http.ServeMux) {
+	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		if prometheusRoute.MatchString(req.URL.Path) {
+			api.prometheusStatus(w, req)
+		} else {
+			w.WriteHeader(404)
+		}
+	})
+}
+
+type objectReference struct {
+	name      string
+	namespace string
+}
+
+func parsePrometheusStatusUrl(path string) objectReference {
+	matches := prometheusRoute.FindAllStringSubmatch(path, -1)
+	ns := ""
+	name := ""
+	if len(matches) == 1 {
+		if len(matches[0]) == 3 {
+			ns = matches[0][1]
+			name = matches[0][2]
+		}
+	}
+
+	return objectReference{
+		name:      name,
+		namespace: ns,
+	}
+}
+
+func (api *API) prometheusStatus(w http.ResponseWriter, req *http.Request) {
+	or := parsePrometheusStatusUrl(req.URL.Path)
```

Comment:
This could just return `(namespace, name string)` rather than a custom type. But doesn't have to be changed now.